src/components/MetricsPane/charts/AggregateBleaching.jsx

```diff
@@ -0,0 +1,136 @@
+import { useContext } from 'react'
+import Plot from 'react-plotly.js'
+
+import { ChartSubtitle, ChartWrapper, TitlesWrapper } from './Charts.styles'
+import { FilterProjectsContext } from '../../../context/FilterProjectsContext'
+import { MetricCardH3 } from '../MetricsPane.styles'
+import dashboardOnlyTheme from '../../../styles/dashboardOnlyTheme'
+
+const chartTheme = dashboardOnlyTheme.plotlyChart
+
+export const AggregateBleaching = () => {
+  const { displayedSurveys } = useContext(FilterProjectsContext)
+
+  const initialColoniesBleachedSummarizedBySeverity = {
+    numColoniesTotal: 0,
+    totalColoniesNormal: 0,
+    totalColoniesPale: 0,
+    totalColonies0to20: 0,
+    totalColonies20to50: 0,
+    totalColonies50to80: 0,
+    totalColonies80to100: 0,
+    totalColoniesDead: 0,
+  }
+  const coloniesBleachedSummarizedBySeverity = displayedSurveys
+    .filter((record) => !!record.protocols?.colonies_bleached)
+    .reduce((accumulator, record) => {
+      const {
+        count_total_avg,
+        percent_100_avg,
+        percent_20_avg,
+        percent_50_avg,
+        percent_80_avg,
+        percent_dead_avg,
+        percent_normal_avg,
+        percent_pale_avg,
+        sample_unit_count,
+      } = record.protocols.colonies_bleached
+      const numColoniesTotalForRecord = count_total_avg * sample_unit_count
+
+      if (isNaN(numColoniesTotalForRecord)) {
+        return accumulator
+      }
+
+      const totalColoniesNormal =
+        (numColoniesTotalForRecord * percent_normal_avg) / 100 +
+        (accumulator.totalColoniesNormal ?? 0)
+      const totalColoniesPale =
+        (numColoniesTotalForRecord * percent_pale_avg) / 100 + (accumulator.totalColoniesPale ?? 0)
+      const totalColonies0to20 =
+        (numColoniesTotalForRecord * percent_20_avg) / 100 + (accumulator.totalColonies0to20 ?? 0)
+      const totalColonies20to50 =
+        (numColoniesTotalForRecord * percent_50_avg) / 100 + (accumulator.totalColonies20to50 ?? 0)
+      const totalColonies50to80 =
+        (numColoniesTotalForRecord * percent_80_avg) / 100 + (accumulator.totalColonies50to80 ?? 0)
+      const totalColonies80to100 =
+        (numColoniesTotalForRecord * percent_100_avg) / 100 +
+        (accumulator.totalColonies80to100 ?? 0)
+      const totalColoniesDead =
+        (numColoniesTotalForRecord * percent_dead_avg) / 100 + (accumulator.totalColoniesDead ?? 0)
+
+      return {
+        numColoniesTotal: numColoniesTotalForRecord + accumulator.numColoniesTotal,
+        totalColoniesNormal,
+        totalColoniesPale,
+        totalColonies0to20,
+        totalColonies20to50,
+        totalColonies50to80,
+        totalColonies80to100,
+        totalColoniesDead,
+      }
+    }, initialColoniesBleachedSummarizedBySeverity)
```

Comment:
@saanobhaai this is definitely worth a review by you to ensure I translated what R was doing properly. We can do a walkthrough in tomorrows meeting if that makes sense.